### PR TITLE
chore: re-enable napi client tests

### PR DESCRIPTION
**Pull request by williamluke4**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -10,35 +10,35 @@ on:
...
+      - "*.md"
+      - ".vscode/**"
+      - "docs/**"
+      - "examples/**"
+      - "LICENSE"
+      - "src/.buildkite/**"
+      - "src/.dockerignore"
+      - "*.bench.ts"
+      - "src/scripts/ci/publish.ts"
+      - "src/graphs/**"
```
Comment (Jolg42) [inline .github/workflows/test.yml:22]: Not sure which plugin/setting is making PRs look like this for this file, I think that my config will probably want to revert the change 😢

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -355,7 +415,7 @@ jobs:
     timeout-minutes: 10
```
Comment (Jolg42) [inline .github/workflows/test.yml:415]: Just noticed we already have `client-types-napi`

Is there any type difference between napi or not?

**Review by Jolg42 — APPROVED** [review]

**Review by Jolg42 — DISMISSED** [review]
Actually please check logs to see if they are a problem or not

I see many
```
console.error
      error: database "tests-blog-env-postgresql" is being accessed by other users
```
https://github.com/prisma/prisma/pull/6743/checks?check_run_id=2418290097

File: .github/workflows/test.yml (hunk 1 above)
Comment (williamluke4) [reply .github/workflows/test.yml:22]: Ahh bugger let me sort that out

File: .github/workflows/test.yml (hunk 2 above)
Comment (williamluke4) [reply .github/workflows/test.yml:415]: No, there shouldn't be! So we should be able to remove that

Comment (williamluke4) [reply .github/workflows/test.yml]: Removed

File: src/packages/engine-core/src/NAPIEngine.ts [hunk 3]
```diff
@@ -458,10 +468,7 @@ You may have to run ${chalk.greenBright(
   ): Promise<{ data: T; elapsed: number }> {
     try {
       await this.start()
-      debug(`request: ${this.connected}`)
-      if (!this.connected) {
-        await this.start()
-      }
+      debug(`state:${this.connected}`)
```
Comment (Jolg42) [inline src/packages/engine-core/src/NAPIEngine.ts:471]:
```suggestion
      debug(`request state: ${this.connected}`)
```

Since it happens in `request`

File: src/packages/engine-core/src/NAPIEngine.ts [hunk 4]
```diff
@@ -421,10 +430,11 @@ You may have to run ${chalk.greenBright(
   }
   async getConfig(): Promise<GetConfigResult> {
     await this.start()
-    debug('getConfig')
-    return this.parseEngineResponse<GetConfigResult>(
+    const config = this.parseEngineResponse<GetConfigResult>(
       await this.engine!.getConfig(),
     )
+    debug(`getConfig: ${JSON.stringify(config, undefined, 2)}`)
```
Comment (Jolg42) [inline src/packages/engine-core/src/NAPIEngine.ts:436]:
```suggestion
```
Probably better to remove to not leak info?

File: src/packages/engine-core/src/NAPIEngine.ts [hunk 5]
```diff
@@ -388,22 +391,28 @@ You may have to run ${chalk.greenBright(
 
   async stop(): Promise<void> {
     await this.connectPromise
-    debug('stop')
+    await this.currentQuery
```
Comment (Jolg42) [inline src/packages/engine-core/src/NAPIEngine.ts:394]: So now we wait for current query to finish?

Comment (williamluke4) [reply src/packages/engine-core/src/NAPIEngine.ts:394]: yep

Comment (Jolg42) [reply src/packages/engine-core/src/NAPIEngine.ts:394]: Are we currently doing that with NodeEngine? (I don't think so?)

Comment (williamluke4) [reply src/packages/engine-core/src/NAPIEngine.ts:394]: https://github.com/prisma/prisma/blob/client/telemetry/src/packages/engine-core/src/NodeEngine.ts#L840

Comment (Jolg42) [reply src/packages/engine-core/src/NAPIEngine.ts:394]: Nice thanks!

**Review by Jolg42 — APPROVED** [review]
Well done!
